src/psyclone/tests/domain/common/extract_driver_creator_test.py

```diff
@@ -86,33 +86,32 @@ def test_driver_creation1(tmpdir):
     default name.
 
     '''
-    # Use tmpdir so that the driver is created in tmp
-    tmpdir.chdir()
-
-    etrans = GOceanExtractTrans()
-    psy, invoke = get_invoke("driver_test.f90",
-                             GOCEAN_API, idx=0, dist_mem=False)
-    schedule = invoke.schedule
-
-    etrans.apply(schedule.children[0], {'create_driver': True})
-    # We are only interested in the driver, so ignore results.
-    str(psy.gen)
-
-    driver = tmpdir.join("driver-psy_extract_example_with_various_variable_"
-                         "access_patterns-invoke_0_compute_kernel:compute_"
-                         "kernel_code:r0.f90")
-    assert driver.isfile()
-
-    with driver.open("r") as driver_file:
-        driver_code = driver_file.read()
-
-    # This is an excerpt of the code that should get created.
-    # It is tested line by line since there is other code in between
-    # which is not important, and the order might also change. It also
-    # tests if unique variable names are created in the driver: the user
-    # program contains a local variable 'dx', which clashes with the grid
-    # property dx. The grid property will be renamed to 'dx_1':
-    expected = '''use read_kernel_data_mod, only : ReadKernelDataType
+    with change_dir(tmpdir):
```

Comment:
Could we do this as a fixture instead?

Reply:
Turns out ... we can. I still needed the `change_dir` function in some places  (e.g. the compilation classes use it to build the infrastructure libs, gocean_opencl_trans_test.py does, ...). And I think it's neat enough to leave it in place and not go back to the old 'os.chdir` way of things.
I could also change quite a few tests that previously used `tmpdir` besides using it in the `chdir` command - quite often it was used to assemble an absolute path to a file, but because we already are in the corresponding directory, I could just use a relative path. Only exception was iirc config_test, which compares the (absolute) path in one or two cases, and I think testing the full path is better, so I used `os.getcwd()` once or twice.

Unfortunately, since `change_dir` is not actually used, I had to add plenty of `pylint disable=unused-argument` :(

Reply:
In that case, the right thing to do is to use `@pytest.mark.usefixtures("my-fixture-name,...")`. Please could you use that rather than disabling the pylint error?

Reply:
Great suggestion, I didn't think of that :)